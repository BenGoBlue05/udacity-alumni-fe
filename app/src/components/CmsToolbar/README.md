## CmsToolbar Component
A component that acts as the main toolbar for the cms.

### Example

```js
<CmsToolbar
  spotlighted={spotlighted}
  onToggleSpotlight={this.setSpotlightToggle}
  onSetStatus={this.setStatus}
  status={status}
/>
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **spotlighted**    | Bool   |  False   | Any boolean value
| **onToggleSpotlight**    | Func   |     | Callback Function for the spotlight toggle
| **onSetStatus**    |  Func   |     | Callback Function for the status toggle
| **status**    | Int / Enum   |  0   | The integer value for the status (0-2)


### Other Information
