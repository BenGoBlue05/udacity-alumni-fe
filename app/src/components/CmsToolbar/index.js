import React, { PropTypes } from 'react';
import styles from './index.module.scss';
import cssModules from 'react-css-modules';
import Columns from 'grommet-udacity/components/Columns';
import Box from 'grommet-udacity/components/Box';
import Button from 'grommet-udacity/components/Button';
import ContractIcon from 'grommet-udacity/components/icons/base/Contract';

const CmsToolbar = ({
  onSave,
  canSubmit,
  onPreview,
  isEditing,
  onToggleVisibility,
  isVisible,
}) => (
  <Box
    align="center"
    justify="center"
    full="horizontal"
    className={styles.toolbar}
  >
    {isVisible &&
      <Columns justify="center" full="horizontal">
        <Box
          direction="row"
          full="horizontal"
          justify="between"
          align="center"
          pad="medium"
          margin="small"
          className={styles.toolbarWrapper}
          colorIndex="light-2"
        >
          <Button
            secondary
            className={styles.button}
            label="Preview"
            onClick={onPreview}
          />
          <Button
            plain
            onClick={onToggleVisibility}
            icon={<ContractIcon />}
          />
          <Button
            primary
            className={styles.button}
            label={isEditing ? 'Update Article' : 'Save Article'}
            onClick={canSubmit ? onSave : null}
          />
        </Box>
      </Columns>
    }
  </Box>
);

CmsToolbar.propTypes = {
  canSubmit: PropTypes.bool.isRequired,
  onPreview: PropTypes.func.isRequired,
  onSave: PropTypes.func.isRequired,
  isEditing: PropTypes.bool.isRequired,
  onToggleVisibility: PropTypes.func.isRequired,
  isVisible: PropTypes.bool.isRequired,
};

export default cssModules(CmsToolbar, styles);
