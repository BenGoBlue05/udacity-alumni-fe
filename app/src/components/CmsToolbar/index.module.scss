.toolbar {
  position: fixed;
  margin-bottom: 0px !important;
  padding: 0px !important;
  left: 0;
  bottom: -10px;
  width: 100%;
  z-index: 100;
}
