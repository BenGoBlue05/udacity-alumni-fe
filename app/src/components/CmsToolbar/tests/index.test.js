import CmsToolbar from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<CmsToolbar />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <CmsToolbar
        onSave={e => e}
        canSubmit={true}
        onPreview={e => e}    
        isEditing={true}
        onToggleVisibility={e => e}  
        isVisible={true}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
