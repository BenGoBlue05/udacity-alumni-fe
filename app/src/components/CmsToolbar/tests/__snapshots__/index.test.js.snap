exports[`<CmsToolbar /> should render with default props 1`] = `
<Box
  align="center"
  announce={false}
  direction="column"
  focusable={true}
  full="horizontal"
  justify="center"
  pad={
    Object {
      "horizontal": "medium",
      "vertical": "medium"
    }
  }
  responsive={true}
  tag="div">
  <Columns
    justify="center"
    maxCount={1}
    responsive={true}>
    <Box
      align="center"
      announce={false}
      colorIndex="light-2"
      direction="row"
      focusable={true}
      justify="between"
      margin="small"
      pad="medium"
      responsive={true}
      size="large"
      tag="div">
      <CheckBox
        checked={false}
        id="spotlight-toggle"
        label="Spotlight"
        name="spotlight"
        onChange={[Function onToggleSpotlight]}
        toggle={true} />
      <Select
        label="Status"
        onChange={[Function onChange]}
        options={
          Array [
            Object {
              "label": "Draft",
              "value": 0
            },
            Object {
              "label": "Published",
              "value": 1
            },
            Object {
              "label": "Archived",
              "value": 2
            }
          ]
        }
        value="Draft" />
    </Box>
  </Columns>
</Box>
`;
